crates/biome_js_analyze/src/lint/nursery/no_useless_undefined_initialization.rs

```diff
@@ -0,0 +1,139 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_syntax::JsVariableDeclaration;
+use biome_rowan::{AstNode, BatchMutationExt, TextRange};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Disallow initializing variables to undefined.
+    ///
+    /// A variable that is declared and not initialized to any value automatically gets the value of undefined.
+    /// It’s considered a best practice to avoid initializing variables to undefined.
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-undef-init
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = undefined;
+    ///
+    /// let b = undefined, c = 1, d = undefined;
+    ///
+    /// for (let i = 0; i < 100; i++) {
+    /// 	let i = undefined;
+    /// }
+    /// ```
```

Comment:
I am not sure if you read the [contribution guide](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md#document-the-rule). Each snippet **must** emit only one diagnostic. 
```suggestion
    /// ```js,expect_diagnostic
    /// var a = undefined;
    /// ```
    /// ```js,expect_diagnostic
    /// let b = undefined, c = 1, d = undefined;
    /// ```
    /// ```js,expect_diagnostic
    /// for (let i = 0; i < 100; i++) {
    /// 	let i = undefined;
    /// }
    /// ```
```